pkg/provisioner/ironic/testserver/server.go

```diff
@@ -76,16 +86,59 @@ func (m *MockServer) Response(pattern string, payload string) *MockServer {
 	return m.ResponseWithCode(pattern, payload, http.StatusOK)
 }
 
+func (m *MockServer) buildHandler(pattern string) func(http.ResponseWriter, *http.Request) {
+
+	handler := func(w http.ResponseWriter, r *http.Request) {
+
+		mh, ok := m.responsesByMethod[r.URL.String()]
```

Comment:
Does this value of `ok` need to be checked?

Reply:
Fixed, no need by design